Library/Homebrew/test/java_requirement_test.rb

```diff
@@ -0,0 +1,11 @@
+require "testing_env"
+require "requirements/java_requirement"
+
+class JavaRequirementTests < Homebrew::TestCase
+  def test_version_s
+    x = JavaRequirement.new(%w[1.8])
+    assert_equal x.display_s, "java = 1.8"
+    y = JavaRequirement.new(%w[1.8+])
+    assert_equal y.display_s, "java >= 1.8"
```

Comment:
A great test? No, but there's frankly not much to test and it's better than nothing IMO.